src/H5Fint.c

```diff
@@ -1515,6 +1542,12 @@ H5F__dest(H5F_t *f, hbool_t flush)
             /* Push error, but keep going*/
             HDONE_ERROR(H5E_FILE, H5E_CANTRELEASE, FAIL, "problems closing file")
 
+        /* If this is a VFD SWMR writer, prep for flush or close */
```

Comment:
The H5F_vfd_swmr_writer_prep_for_flush_or_close() code revolves around the page buffer and therefore, this 'if' should be move to the beginning of H5PB_dest().